src/main/scala/mesosphere/marathon/api/AuthResource.scala

```diff
@@ -16,15 +19,23 @@ trait AuthResource extends RestResource {
 
   def authenticated(request: HttpServletRequest)(fn: Identity => Response): Response = {
     val requestWrapper = new RequestFacade(request)
-    val maybeIdentity = result(authenticator.authenticate(requestWrapper))
-    maybeIdentity.map { identity =>
-      try {
-        fn(identity)
-      } catch {
-        case e: AccessDeniedException => withResponseFacade(authorizer.handleNotAuthorized(identity, _))
-      }
-    }.getOrElse {
-      withResponseFacade(authenticator.handleNotAuthenticated(requestWrapper, _))
+    val authenticationRequest = authenticator.authenticate(requestWrapper)
+
+    Await.ready(authenticationRequest, config.zkTimeoutDuration)
+
+    authenticationRequest.value.get match {
```

Comment:
I am not super happy with the `.get` here, but my scala-fu failed me here... any suggestions?